cms/djangoapps/maintenance/views.py

```diff
@@ -0,0 +1,209 @@
+            "commands": MAINTENANCE_COMMANDS,
+        })
+
+
+class MaintenanceBaseView(View):
+
+    template = 'maintenance/container.html'
+
+    def __init__(self, command=None):
+        self.context = {
+            'command': command if command else '',
+            'form_data': {},
+            'error': False,
+            'msg': ''
+        }
+
+    def render_response(self):
+        return render_to_response(self.template, self.context)
+
+    @method_decorator(require_global_staff)
+    def get(self, request):
+        """Render get view."""
+        return self.render_response()
+
+    def validate_course_key(self, course_key, branch=None):
+        """Validates course_key and updates context, returns usage_key, that would be used by maintenance app views."""
+        course_usage_key = None
+        if not course_key:
+            self.context['error'] = True
+            self.context['msg'] = COURSE_KEY_ERROR_MESSAGES['empty_course_key']
+            return course_usage_key
+        try:
+            if branch == "published":
+                course_key += "+branch@published-branch"
+
+            course_usage_key = CourseKey.from_string(course_key)
+            if not modulestore().has_course(course_usage_key):
+                raise ItemNotFoundError
+
+            self.context.update({
+                'error': False,
+                'msg': ''
+            })
+        except InvalidKeyError:
+            self.context['error'] = True
+            self.context['msg'] = COURSE_KEY_ERROR_MESSAGES['invalid_course_key']
+        except ItemNotFoundError:
+            self.context['error'] = True
+            self.context['msg'] = COURSE_KEY_ERROR_MESSAGES['course_key_not_found']
+        return course_usage_key
+
+
+class ForcePublishCourseView(MaintenanceBaseView):
+    """
+    View for force publish state of the course, used by the escalation team.
+    """
+
+    def __init__(self):
+        super(ForcePublishCourseView, self).__init__(MAINTENANCE_COMMANDS['force_publish_course'])
+        self.context.update({
+            'current_versions': [],
+            'updated_versions': [],
+            'form_data': {
+                'course_id': '',
+                'is_dry_run': True
+            }
+        })
+
+    @transaction.atomic
+    @method_decorator(require_global_staff)
+    def post(self, request):
+        """Force publish a course."""
+
+        course_id = request.POST.get('course-id')
+        is_dry_run = bool(request.POST.get('dry-run'))
+
+        course_usage_key = self.validate_course_key(course_id)
+
+        self.context.update({
+            'form_data': {
+                'course_id': course_id,
+                'is_dry_run': is_dry_run
+            }
+        })
+
+        if self.context['error']:
+            return self.render_response()
+
+        owning_store = modulestore()._get_modulestore_for_courselike(course_usage_key)  # pylint: disable=protected-access
```

Comment:
`owning_store` is the modulestore of the course. Like if we have passed `course-v1:edx+DemoX+Demo` it would be split modeulestore if course key is `edx/XSS/101` it would be mongo modulestore.

We are checking this because, force publish does not work for mongo courses.